### PR TITLE
Ran linting and added configuration for Black, flake8 and isort.

### PR DESCRIPTION
**Pull request by pboers1988**
Added best practice configuration for black, isort, flake8 mypy and pre-commit. Edited all the files make sure they conform.

### CODE REVIEW

**Review by lunkwill42 — COMMENTED** [review]
I think this is a great and much needed effort, especially considering future collaborative effort.

I would also argue that this effort should include a `.git-blame-ignore-revs` file, so that `git blame` can ignore automated formatting changes when resolving who-did-what: https://michaelheap.com/git-ignore-rev/

We have made use of this to great success when automatically reformatting all of our NAV codebase, a codebase that was much larger and older than CNaaS-NMS :)

**Comment by indy-independence** [comment]
I'm thinking maybe now is a good time to do this switch to black
Is the pre-commit working in this branch or do we need a file .pre-commit-config.yaml ?

**Comment by lunkwill42** [comment]
> I'm thinking maybe now is a good time to do this switch to black Is the pre-commit working in this branch or do we need a file .pre-commit-config.yaml ?

We need a `.pre-commit-config.yaml` to specify which hooks to actually run when `git commit` is run. Not sure if @pboers1988 ran black/isort/flake manually here, or if he just forgot to commit the `.pre-commit-config.yaml` that specified them as hooks.

Regardless, I think this is old enough to warrant a rebase and re-run all the tools on the codebase over again (maybe just keep the configuration commit and give the tool run commits a do-over).

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+[tool.isort]
+profile="black"
+line_length = 120
+skip = ["venv"]
+
+[tool.black]
+line-length = 120
```
Comment (lunkwill42) [inline pyproject.toml:7]: Actually, the contributing guide for CNaaS-NMS specifies a line-length of 99 as policy. This should comply with that (unless SUNET would like to change their policy).

**Comment by indy-independence** [comment]
Integrated these changes into #279

**Pull request closed without merging**